sktime/forecasting/prophet.py

```diff
@@ -0,0 +1,341 @@
+        range of days around the date to be included as holidays.
+        lower_window=-2 will include 2 days prior to the date as holidays. Also
+        optionally can have a column prior_scale specifying the prior scale for
+        that holiday.
+    seasonality_mode: 'additive' (default) or 'multiplicative'.
+    seasonality_prior_scale: Parameter modulating the strength of the
+        seasonality model. Larger values allow the model to fit larger seasonal
+        fluctuations, smaller values dampen the seasonality. Can be specified
+        for individual seasonalities using add_seasonality.
+    holidays_prior_scale: Parameter modulating the strength of the holiday
+        components model, unless overridden in the holidays input.
+    changepoint_prior_scale: Parameter modulating the flexibility of the
+        automatic changepoint selection. Large values will allow many
+        changepoints, small values will allow few changepoints.
+    mcmc_samples: Integer, if greater than 0, will do full Bayesian inference
+        with the specified number of MCMC samples. If 0, will do MAP
+        estimation.
+    interval_width: Float, width of the uncertainty intervals provided
+        for the forecast. If mcmc_samples=0, this will be only the uncertainty
+        in the trend using the MAP estimate of the extrapolated generative
+        model. If mcmc.samples>0, this will be integrated over all model
+        parameters, which will include uncertainty in seasonality.
+    uncertainty_samples: Number of simulated draws used to estimate
+        uncertainty intervals. Settings this value to 0 or False will disable
+        uncertainty estimation and speed up the calculation.
+    stan_backend: str as defined in StanBackendEnum default: None - will try to
+        iterate over all available backends and find the working one
+
+    References
+    ----------
+    https://facebook.github.io/prophet
+    https://github.com/facebook/prophet
+
+    """
+
+    def __init__(
+        self,
+        # Args due to wrapping
+        freq,
+        add_seasonality=None,
+        add_country_holidays=None,
+        # Args of fbprophet
+        growth='linear',
+        changepoints=None,
+        n_changepoints=25,
+        changepoint_range=0.8,
+        yearly_seasonality='auto',
+        weekly_seasonality='auto',
+        daily_seasonality='auto',
+        holidays=None,
+        seasonality_mode='additive',
+        seasonality_prior_scale=10.0,
+        holidays_prior_scale=10.0,
+        changepoint_prior_scale=0.05,
+        mcmc_samples=0,
+        interval_width=1 - DEFAULT_ALPHA,
+        uncertainty_samples=1000,
+        stan_backend=None,
+        **kwargs
+    ):
+        self.freq = freq
+        self.add_seasonality = add_seasonality
+        self.add_country_holidays = add_country_holidays
+
+        self.growth = growth
+        self.changepoints = changepoints
+        if self.changepoints is not None:
```

Comment:
I'd wrap these checks into a function - something like `_check_changepoints` - call it in fit, and assign the return value to self.

Reply:
done